### DISCUSSION TITLE
Request for entryType data attribute in Control Panel entry edit view

### DISCUSSION THREAD

**Discussion by watershed** [craftcms/cms#12114] [Ideas] [answered]
Today I wanted to use the [Control Panel CSS plugin](https://plugins.craftcms.com/cp-css) to apply something specific subject to `entryType` but have concluded that without further work* I can’t, because there is no HTML hook in an entry’s edit view with which to do so.

I realise that `entryType` is a value which can be edited, but it would be nice if the currently selected value could be written to the DOM with something like:

    <form id="main-form" … data-entry-type="{{ entryType }}" …>

\* I guess I can use JavaScript to:

    const currentEntryType = document.querySelector('#entryType option[selected]');

…and go from there. (Fingers crossed that Control Panel JS runs before Control Panel CSS.)

**Comment by watershed** [comment, marked as answer]
Actually, using [Control Panel JS](https://plugins.craftcms.com/cp-js) gets the job done just fine, so maybe this request is redundant.

Here is what I did to create the hook I needed:

```
const savedEntryType = document.querySelector('#entryType option[selected]') || null;

if ( savedEntryType ) {
    const eType = savedEntryType.textContent;
    document.querySelector('#main-form').setAttribute('data-saved-entry-type', eType);
}
```

**Reply by brandonkelly** [reply]
Considering the page doesn’t refresh when the entry type is changed, you might want to add a `change` listener to the input and update the `data-saved-entry-type` attribute.

**Reply by watershed** [reply]
Good thought. Thanks Brandon.
